File: .github/workflows/PHP/sportsonline.php
<?php
$url = "https://sportsonline.gl/prog.txt";

$text = file_get_contents($url);

// Regular expression to match the pattern (time, event, URL)
$pattern = '/(\d{2}:\d{2})\s+(.*?)\s+\|\s+(https?:\/\/[^\s]+)/';

// Extract matches
preg_match_all($pattern, $text, $matches, PREG_SET_ORDER);

// Create an array to store results
$result = [];

date_default_timezone_set("Asia/Bangkok");

foreach ($matches as $match) {
    // Parse the time and add 7 hours
    $original_time = $match[1];
    $newTime = date("H:i", strtotime($original_time . " +7 hours"));
    
    // Create the updated name with the new time
    $updatedName = $newTime . " " . $match[2];
    
    $result[] = [
        'name' => $updatedName,
        'image' => 'https://static.thairath.co.th/media/dFQROr7oWzulq5Fa5LJEXNHgK8Nkou0mFfNIBOUFttbzB1Odu55E3VaV0f8gNyuUy8L.jpg',
        'url' => $match[3]
    ];
}

// Convert the array to JSON
$json = json_encode($result, JSON_PRETTY_PRINT);

// Output the JSON
echo '{ "name": "<a href='.$url .' target=_blank>sportsonline</a>",<br>';
echo '"image": "https://static.thairath.co.th/media/dFQROr7oWzulq5Fa5LJEXNHgK8Nkou0mFfNIBOUFttbzB1Odu55E3VaV0f8gNyuUy8L.jpg",<br>';
echo '"author": "' . 'update' . ' ' . date("d/m/Y H:i") . '",<br>';
echo '"url": "", <br> "stations":  <br>';
// Output the JSON
echo $json;
echo '}';
?>

File: .github/workflows/PYTHON/sportsonline.py
import re
import requests
import json
from datetime import datetime, timedelta

# URL of the text file
url = "https://sportsonline.gl/prog.txt"

# Fetch the content of the text file
response = requests.get(url)
if response.status_code != 200:
    print("Failed to fetch the content from the URL.")
    exit()

# Searching for time pattern in the content of the file
matches = re.findall(r"(\d{2}:\d{2}.{3}.*?) \| (http.*?\.php)", response.text)

# Get today's date in the required format
today_date = datetime.now().strftime("%d/%m/%Y %H:%M:%S")

# Template for the final JSON structure
data = {
    "name": "sportsonline.gl",
    "image": "https://static.thairath.co.th/media/dFQROr7oWzulq5Fa5LJEXNHgK8Nkou0mFfNIBOUFttbzB1Odu55E3VaV0f8gNyuUy8L.jpg",
    "url": "https://dl.dropbox.com/scl/fi/04bcea7wnj2tea0bbqtuj/SPORT.w3u?rlkey=fogj7jaec08k2klrxjl8xdn3h&st=ho0u09uh&dl=0",
    "author": f"royrover update {today_date}",
    "stations": []
}

# Populate the "stations" list with extracted matches
image_url = "https://static.thairath.co.th/media/dFQROr7oWzulq5Fa5LJEXNHgK8Nkou0mFfNIBOUFttbzB1Odu55E3VaV0f8gNyuUy8L.jpg"
referer = "https://sportsonline.gl"

for match in matches:
    event_name, event_url = match
    # Extract and add 6 hours to the time in event_name
    time_match = re.match(r"(\d{2}:\d{2})", event_name)
    if time_match:
        original_time = datetime.strptime(time_match.group(1), "%H:%M")
        updated_time = original_time + timedelta(hours=6)
        event_name = event_name.replace(time_match.group(1), updated_time.strftime("%H:%M"))

    # Append the event to the stations list
    data["stations"].append({
        "name": event_name.strip(),
        "image": image_url,
        "url": event_url,
        "isHost": "true",
        "referer": referer
    })

# Write the JSON data to SPORT.w3u file
with open("SPORT.w3u", "w") as file:
    file.write(json.dumps(data, indent=4))

print("File SPORT.w3u created successfully.")
